e2e_test/source_inline/kafka/avro/name_strategy.slt

```diff
@@ -0,0 +1,203 @@
+control substitution on
+
+system ok
+rpk topic delete 'upsert_avro_json.*'
+
+
+# wrong strategy name
+statement error expect strategy name in topic_name_strategy, record_name_strategy and topic_record_name_strategy
+create source s1 () with (
+  connector = 'kafka',
+  topic = 'upsert_avro_json-record',
+  properties.bootstrap.server = '${RISEDEV_KAFKA_BOOTSTRAP_SERVERS}'
+) format plain encode avro (
+  schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}',
+  schema.registry.name.strategy = 'no sense',
+  message = 'CPLM.OBJ_ATTRIBUTE_VALUE',
+);
+
+
+#######################
+# topic_name_strategy
+#######################
+
+# TODO: refactor the producer script and the test data format.
+system ok
+python3 scripts/source/schema_registry_producer.py  "${RISEDEV_KAFKA_BOOTSTRAP_SERVERS}" "${RISEDEV_SCHEMA_REGISTRY_URL}" e2e_test/source_inline/kafka/avro/upsert_avro_json "topic" "avro"
+
+# If we cannot extract key schema, use message key as varchar primary key
+statement ok
+CREATE TABLE t_topic ( primary key (rw_key) )
+INCLUDE KEY AS rw_key
+WITH (
+	connector = 'kafka',
+  properties.bootstrap.server = '${RISEDEV_KAFKA_BOOTSTRAP_SERVERS}',
+	topic = 'upsert_avro_json')
+FORMAT UPSERT ENCODE AVRO (schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}');
+
+
+# TODO: Uncomment this when we add test data kafka key with format `"ID":id`
+# statement ok
+# CREATE TABLE upsert_avro_json (
+#   PRIMARY KEY("ID")
+# )
+# WITH (
+# connector = 'kafka',
+#  properties.bootstrap.server = '${RISEDEV_KAFKA_BOOTSTRAP_SERVERS}',
+# topic = 'upsert_avro_json')
+# FORMAT UPSERT ENCODE AVRO (schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}');
```

Comment:
I don't actually understand what's the commented test mean. Just kept it

Reply:
@yuhao-su  https://github.com/risingwavelabs/risingwave/pull/13023 Do you still remember why we commented out the test case?

Reply:
We used to parse kafka key with Json format. That's way it required the key looks like `"ID": id`. But now we parse kafka key as `bytes`, so I think we can uncomment this now.

Reply:
try enable in https://github.com/risingwavelabs/risingwave/pull/17062